lightning/src/events/bump_transaction.rs

```diff
@@ -228,3 +239,359 @@ pub enum BumpTransactionEvent {
+	}
+}
+
+impl<W: Deref> UtxoSource<&CoinSelectionSource, W> where W::Target: WalletSource {
+	pub fn from_wallet(w: W) -> Self {
+		Self::Wallet(w)
+	}
+}
+
+impl<C: Deref, W: Deref> UtxoSource<C, W>
+where
+	C::Target: CoinSelectionSource,
+	W::Target: WalletSource,
+{
+	fn sign_confirmed_utxo(&self, tx: &Transaction, input: usize) -> Result<Witness, ()> {
+		match self {
+			UtxoSource::CoinSelection(source) => source.sign_confirmed_utxo(tx, input),
+			UtxoSource::Wallet(source) => source.sign_confirmed_utxo(tx, input),
+		}
+	}
+}
+
+pub struct BumpTransactionEventHandler<B: Deref, C: Deref, W: Deref, SP: Deref, L: Deref>
+where
+	B::Target: BroadcasterInterface,
+	C::Target: CoinSelectionSource,
+	W::Target: WalletSource,
+	SP::Target: SignerProvider,
+	L::Target: Logger,
+{
+	broadcaster: B,
+	utxo_source: UtxoSource<C, W>,
+	signer_provider: SP,
+	logger: L,
+	secp: Secp256k1<secp256k1::All>,
+}
+
+impl<B: Deref, C: Deref, W: Deref, SP: Deref, L: Deref> BumpTransactionEventHandler<B, C, W, SP, L>
+where
+	B::Target: BroadcasterInterface,
+	C::Target: CoinSelectionSource,
+	W::Target: WalletSource,
+	SP::Target: SignerProvider,
+	L::Target: Logger,
+{
+	pub fn new(broadcaster: B, utxo_source: UtxoSource<C, W>, signer_provider: SP, logger: L) -> Self {
+		Self {
+			broadcaster,
+			utxo_source,
+			signer_provider,
+			logger,
+			secp: Secp256k1::new(),
+		}
+	}
+
+	pub fn handle_event(&self, event: Event) {
+		let event = if let Event::BumpTransaction(event) = event {
+			event
+		} else {
+			return;
+		};
+
+		match event {
+			BumpTransactionEvent::ChannelClose {
+				package_target_feerate_sat_per_1000_weight, commitment_tx, anchor_descriptor, ..
+			} => {
+				if let Err(_) = self.handle_channel_close(
+					package_target_feerate_sat_per_1000_weight, &commitment_tx, &anchor_descriptor,
+				) {
+					log_error!(self.logger, "Failed bumping commitment transaction fee for {}",
+						commitment_tx.txid());
+				}
+			}
+			BumpTransactionEvent::HTLCResolution {
+				target_feerate_sat_per_1000_weight, htlc_descriptors, tx_lock_time,
+			} => {
+				if let Err(_) = self.handle_htlc_resolution(
+					target_feerate_sat_per_1000_weight, &htlc_descriptors, tx_lock_time,
+				) {
+					log_error!(self.logger, "Failed bumping HTLC transaction fee for commitment {}",
+						htlc_descriptors[0].commitment_txid);
+				}
+			}
+		}
+	}
+
+	fn handle_channel_close(
+		&self, package_target_feerate_sat_per_1000_weight: u32, commitment_tx: &Transaction,
+		anchor_descriptor: &AnchorDescriptor,
+	) -> Result<(), ()> {
+		let anchor_input = TxIn {
+			previous_output: anchor_descriptor.outpoint,
+			script_sig: Script::new(),
+			sequence: Sequence::ZERO,
+			witness: Witness::new(),
+		};
+		let mut anchor_tx = Transaction {
+			version: 2,
+			lock_time: PackedLockTime::ZERO, // TODO: Use next best height.
+			input: vec![anchor_input],
+			output: vec![],
+		};
+
+		match &self.utxo_source {
+			UtxoSource::CoinSelection(source) => {
```

Comment:
if we provide an interface plugging directly with Bitcoin Core's `fundrawtransaction` and other wallet RPC calls i think we should look that the coin selection abstractions in core match a bit ours, nice if we don't have to re-implement a lot of the confidentiality hardening that has been done there and can rely on their assumptions

Reply:
So I think I do have 2 set of concerns:
- First one breaking some underlying coin selection algorithms privacy hardening if we don’t consume all the set of inputs in the CPFP, e.g see the `CoinSelectionParams::m_avoid_partial_spends` option in `src/wallet/coinselection.h`.
-  Second one accepting “naive” UTXOs from the `CoinSelectionSource` because we don’t have sufficient checks beyond `must_spend` like requesting some depth (e.g at least 3 blocks) and a shallow reorg sweeping out our chain of transactions. On the other side we can consider the responsibility of the coin selection algorithms to give us UTXO of some quality, e.g `fundrawtransaction` has `min_conf` param